cmd/ufs/ufs.go

```diff
@@ -26,8 +28,24 @@ func main() {
 		log.Fatalf("Listen failed: %v", err)
 	}
 
-	s, err := ufs.NewUFS(func(s *protocol.Server) error {
-		s.Trace = nil // log.Printf
+	fs, err := ufs.NewServer(ufs.Root(*root), func(fs *ufs.FileServer) error {
+		if *debug {
+			return ufs.Trace(log.Printf)(fs)
+		}
+
+		return nil
+	})
+
+	var ninefs protocol.NineServer = fs
+	if *debug {
+		ninefs = fs.Debug()
```

Comment:
This just seems a bit awkward to me. You get a server back, the reach into it and get a debug server? Why not just have a debug.NewServer that takes a server?

Reply:
Then the pattern becomes one in which servers can wrap servers. To me, the ultimate test: can a debug server wrap a debug server?

Reply:
I agree, see my latest changes that split the debug server off into a separate package and add a `NewServer(NineServer)` method.